templates/cluster-monitoring-resources.yaml

```diff
@@ -0,0 +1,53 @@
+{{- range $configKey, $config := .Values.configs }}
+{{- if and $config.clusterMonitoring (eq $config.clusterMonitoring.enabled true) }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: prometheus-k8s
+  namespace: {{ $config.namespace }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - services
+  - endpoints
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: prometheus-k8s
+  namespace: {{ $config.namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: prometheus-k8s
+subjects:
+- kind: ServiceAccount
+  name: prometheus-k8s
+  namespace: openshift-monitoring
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ $config.clusterMonitoring.name }}   
+  name: {{ $config.clusterMonitoring.name }}-server-monitor
```

Comment:
@JasonPowr instead of clusterMonitoring.name can we instead use namespace? Then we won't need to add the clusterMonitoring.name to the chart (since it's already within the component) - ideally we would use service or serviceaccount name but as/is the templates don't have access to upstream chart fields - so namespace works for now - wdyt?

Reply:
lgtm!